File: Api/SaveService.cs
namespace Api
{
    public static class SaveService
    {
        public static Record Save(Record record)
        {
            if (record.Check3 == true)
            {
                record.Status = "CLOSED";
            } else if (record.Check2 == true)
            {
                record.Status = "Awaiting Sign Off";
            } else if (record.Check1 == true)
            {
                record.Status = "In Progress";
            }
            return record;
        }
    }
}

File: Site/src/ServiceWorker/Save/save.service.ts
export class SaveService {
    public static save(record: any) : any {
        if(record.check3 === true) {
            record.status = 'CLOSED';
        } else if(record.check2 === true) {
            record.status = 'Awaiting Sign Off';
        } else if(record.check1 === true) {
            record.status = 'In Progress';
        }
        return record;
    }
}
